CMakeLists.txt

```diff
@@ -222,23 +222,23 @@ if(CMAKE_BUILD_TYPE STREQUAL "Debug")
     endif()
   endif()
 elseif(CMAKE_BUILD_TYPE STREQUAL "Release")
-  if(("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU") AND (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 6.2))
-    message(STATUS "GCC >6.2.0 breaks -O3, setting -O2")
+  if(("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU") AND (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7.5))
+    message(STATUS "GCC >7.5.0 breaks -O3, setting -O2")
```

Comment:
Why is this being changed? Do versions from 6.2 to 7.4 not have problems with -O3 after all?

Reply:
Both gcc 6 and 7 seem to got a minor update which fixes it.
I tested a few versions:
- gcc 7.5.0 works (current version on GH runner)
- gcc 7.3.0 fails
- gcc 6.5.0 works
I think we should not differentiate between minor versions. If someone uses gcc 7 it should be (at least) 7.5.0 if it is in a proper package system.